File: projects/mynoc-remootio-angular/src/lib/services/remootioDeviceConfig.ts
import { ErrorObserver, NextObserver } from "rxjs";
import { ReceivedEncryptedFrameContent, RemootioActionResponse } from "./remootioFrames";

export interface IRemootioDeviceConfig {
  deviceIp: string;
  apiSecretKey: string;
  apiAuthKey: string;
  sendPingMessageEveryXMs?: number;
  autoReconnect?: boolean;
  openObserver?: NextObserver<Event>;
  closeObserver?: NextObserver<CloseEvent>;
  messageObserver?: NextObserver<RemootioActionResponse>;
  errorObserver?: ErrorObserver<string>;
  authenticatedObserver?: NextObserver<ReceivedEncryptedFrameContent>;
}

File: dotnet/myNOC.Remootio/RemootioDeviceConfig.cs
namespace myNOC.Remootio;

public sealed class RemootioDeviceConfig
{
    public string DeviceIp { get; set; } = string.Empty;
    public string ApiSecretKey { get; set; } = string.Empty;
    public string ApiAuthKey { get; set; } = string.Empty;
    public int SendPingMessageEveryXMs { get; set; } = 60000;
    public bool AutoReconnect { get; set; } = true;
    public string GateImageUrl { get; set; } = string.Empty;
}
